datafusion/functions/src/datetime/to_char.rs

```diff
@@ -215,8 +219,19 @@ fn _to_char_array(args: &[ColumnarValue]) -> Result<ColumnarValue> {
         };
         // this isn't ideal but this can't use ValueFormatter as it isn't independent
         // from ArrayFormatter
-        let formatter = ArrayFormatter::try_new(arrays[0].as_ref(), &format_options)?;
-        let result = formatter.value(idx).try_to_string();
+        let result = match format {
+            Some(_) => {
+                let formatter =
+                    ArrayFormatter::try_new(arrays[0].as_ref(), &format_options)?;
+                formatter.value(idx).try_to_string()
+            }
+            None => {
+                let null_array = ColumnarValue::create_null_array(1).into_array(1)?;
+                let formatter =
+                    ArrayFormatter::try_new(null_array.as_ref(), &format_options)?;
+                formatter.value(0).try_to_string()
+            }
+        };
```

Comment:
Something else I noticed while looking at this code is that 

```
        ColumnarValue::Array(_) => Ok(ColumnarValue::Array(Arc::new(StringArray::from(
            results,
        )) as ArrayRef)),
```

Effectively means the strings are copied twice (once to `results` and then again to the array).

As a follow on PR we could potentially use `StringBuilder` to build the final string array directly rather than allocating a bunch of small strings.

Reply:
For someone whose role is to find ways to optimize things I seem to be rather poor at it sometimes :) Nice catch.

Reply:
Well there are different degrees of optimization for sure ;)